instrumentation/kamon-akka/src/common/scala/kamon/instrumentation/akka/AkkaMetrics.scala

```diff
@@ -85,14 +87,16 @@ object AkkaMetrics {
     description = "Counts the number of processing errors experienced by the routees of a router"
   )
 
-  def forRouter(path: String, system: String, dispatcher: String, routerClass: String, routeeClass: String): RouterInstruments =
-    new RouterInstruments(TagSet.builder()
+  def forRouter(path: String, system: String, dispatcher: String, routerClass: Class[_], routeeClass: String): RouterInstruments = {
+    val tags = TagSet.builder()
       .add("path", path)
       .add("system", system)
       .add("dispatcher", dispatcher)
-      .add("routerClass", routerClass)
       .add("routeeClass", routeeClass)
-      .build())
+    if (!ActorCellInfo.isTyped(routerClass)) tags.add("routerClass", routerClass.getName)
```

Comment:
Shouldn't this be the routee rather than the router class?

Reply:
Good catch, both the router and the routee class can be "ActorAdapter". I've pushed a small fix for that

But this specific case, no, it should be router. `routerClass` is the tag I'm refactoring